[Ajay Shanker] iP

AjayShanker-geek opened the pull request.

AjayShanker-geek committed:
Rename Chatbot from Duke to Jarvis

AjayShanker-geek committed:
Add method to print breakline and exit

AjayShanker-geek committed:
:tada: Level-0

- Completed Level-0 Assignment
- Rename README info

AjayShanker-geek committed:
:memo: Level-0

- Add comments to methods
- Improve access modifiers and methods

AjayShanker-geek committed:
:art: Level-0

- Improve logo

AjayShanker-geek committed:
Reformat code

AjayShanker-geek committed:
Add Scanner

- Level 1: Echo Task

AjayShanker-geek committed:
Add method to echo user input

- Cleanup messages
- Add Scanner to read user input

AjayShanker-geek committed:
Revert from using indention

- Add print promt as arrow

AjayShanker-geek committed:
Support adding and showing list

- Task: Level-2 Add,List

AjayShanker-geek committed:
:sparkles: Add Task Class

- simple task that has description and done flag

AjayShanker-geek committed:
Rename items to tasks and cleanup

- Remove names from previous tasks

AjayShanker-geek committed:
Add more methods to Task

AjayShanker-geek committed:
Switch to switch to support more commands

- Task Level-3: Add support to mark & unmark

AjayShanker-geek committed:
:bulb: Add comments and cleanup

AjayShanker-geek committed:
Cleanup variable inconsistancy

AjayShanker-geek committed:
Formate Code

AaronZZ10 commented on src/main/java/Jarvis.java:
```diff
@@ -0,0 +1,192 @@
+        printBreakLine();
+    }
+
+    /**
```
the comment clearly states the purpose and parameters of the functions

AaronZZ10 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @return status icon of the task
+     */
+    public String getStatusIcon() {
```
well use of conditional operator to return status icon

AaronZZ10 commented on src/main/java/Jarvis.java:
```diff
@@ -0,0 +1,192 @@
+        System.out.print("  ");
+    }
+
+    /**
```
defining different methods of printing give your code great readability

AaronZZ10 commented on src/main/java/Jarvis.java:
```diff
@@ -0,0 +1,192 @@
+     *
+     * @param status
+     */
+    private static void exit(int status) {
```
I like how you use system.exit to terminate the program

AjayShanker-geek committed:
Update Task
- Level 4 requirement

AjayShanker-geek committed:
Add Todo Class

AjayShanker-geek committed:
Add Dealine Class

AjayShanker-geek committed:
Add Event

AjayShanker-geek committed:
Update Javis
- Move non-related methods
- Clean up switch case

AjayShanker-geek committed:
Improve readability
- Remove commented code

AjayShanker-geek committed:
Update `runtest.sh` with correct chatbot name

AjayShanker-geek committed:
Add Input Test Cases

AjayShanker-geek committed:
Update Expected output

AjayShanker-geek committed:
Add comments and improve code quality

AjayShanker-geek committed:
Create JarvisException

AjayShanker-geek committed:
Add class to handle errors

AjayShanker-geek committed:
Add task catch expection
- Check if description is empty

AjayShanker-geek committed:
Add Todo, Event and Deadline to catch expections
- Check if description is empty
- Check if Event and Deadline with invalid command

AjayShanker-geek committed:
Add Error class to handle error messages and code

AjayShanker-geek committed:
Catch exceptions

AjayShanker-geek committed:
Merge branch 'branch-Level-5'

Allow the programme to catch exceptions

AjayShanker-geek committed:
Divide task classes

AjayShanker-geek committed:
Divide expection classes

AjayShanker-geek committed:
Update runtest script

AjayShanker-geek committed:
Merge branch 'branch-A-Packages'

AjayShanker-geek committed:
Clean up catching exceptions

AjayShanker-geek committed:
Merge branch 'A-Exceptions'

AjayShanker-geek committed:
Update Readme

AjayShanker-geek committed:
Add support task as ArrayList
- created function overload

AjayShanker-geek committed:
Add support task as ArrayList
- created function overload

AjayShanker-geek committed:
Add support to delete task
- Level 6. Delete

AjayShanker-geek committed:
Add saving and loading of tasks

AjayShanker-geek committed:
Add catch exceptions index out of bound

AjayShanker-geek committed:
Add Class to save and load

AjayShanker-geek committed:
Merge branch 'branch-Level-6'

- Merge deletion of task

AjayShanker-geek committed:
Merge branch 'branch-Level-7'

AjayShanker-geek committed:
Add checks if the data file exists

AjayShanker-geek committed:
Improve test cases

AjayShanker-geek committed:
Create jar manifest

yuanners99 commented on src/main/java/org/ajay/exceptions/Error.java:
```diff
@@ -0,0 +1,34 @@
+    }
+}
+
+// public enum ErrorLevel {
```
Remember to remove the commented code.

yuanners99 reviewed (commented):
Nice work! Your code is clean and follows standards well. Keep it up!

AjayShanker-geek committed:
Abstaract prompt relate as Prompt Class

AjayShanker-geek committed:
Remove wild card import

AjayShanker-geek committed:
Clean up main after abstraction Prompt class

AjayShanker-geek committed:
Convert and move exit command to prompt

AjayShanker-geek committed:
Improve runtest.sh script

- script ignore whitespace

AjayShanker-geek committed:
Clean Task Class and Add TaskList

AjayShanker-geek committed:
Add Parser and Constants

AjayShanker-geek committed:
Rename Prompt to Ui

AjayShanker-geek committed:
Cleanup Main

AjayShanker-geek committed:
Restructure code to make the code more OOP

AjayShanker-geek committed:
Move exception into data folder

AjayShanker-geek committed:
Add TaskList and move into data folder

AjayShanker-geek committed:
Split each command into OOP

AjayShanker-geek committed:
Cleanup exceptions catches

AjayShanker-geek committed:
Remove variable to keep track of number of task

AjayShanker-geek committed:
Update text-ui-test

AjayShanker-geek committed:
Merge pull request #1 from AjayShanker-geek/moreOOP

More OOP

- Refactor the code to extract out closely related code as classes.

AjayShanker-geek committed:
Formate and delete unused codes

AjayShanker-geek committed:
Add find command using stream

- Update print list using stream

AjayShanker-geek committed:
Merge branch 'branch-Level-9'

AjayShanker-geek committed:
Ensure storage file and folder exist

AjayShanker-geek committed:
Add support to catch invalid commands

AjayShanker-geek committed:
Add support to catch deadline is not found

AjayShanker-geek committed:
Add support to catch event keyword not found

AjayShanker-geek committed:
Add support to catch out of bound index

AjayShanker-geek committed:
Add support catch empty datetime

AjayShanker-geek committed:
Improve test cases to include edge cases

AjayShanker-geek committed:
Merge branch 'fix/exceptions'

AjayShanker-geek committed:
Clean up UI

AjayShanker-geek committed:
Merge branch 'feature/clean-up'

Cleanup UI

AjayShanker-geek committed:
Add JavaDoc for entry of the program

AjayShanker-geek committed:
Add JavaDoc for storage

AjayShanker-geek committed:
Add JavaDoc Tasks and Commands

AjayShanker-geek committed:
Merge branch 'branch-JavaDoc'

AjayShanker-geek committed:
Update Docs readme

AjayShanker-geek committed:
Merge branch 'branch-UserGuide'

Update Docs readme

AjayShanker-geek committed:
Update runtest

AjayShanker-geek committed:
Update image path

AjayShanker-geek committed:
Fix image path

AjayShanker-geek committed:
Resize image

AjayShanker-geek committed:
Reduce size of the image

AjayShanker-geek committed:
Clean up docs

AjayShanker-geek committed:
Intergrated docsify

AjayShanker-geek committed:
Update User guide info

AjayShanker-geek committed:
Update User Guide image path

AjayShanker-geek committed:
Fix jarvis cli image not showing

AjayShanker-geek committed:
Update repo README.md
